## Confidence Intervals

The goal of this notebook is for the student to be able to construct and interpret confidence intervals using the  StatsModels library.

In [1]:
import scipy
import numpy as np
import matplotlib.pyplot as plt

%config InlineBackend.figure_format = 'retina'

### Confidence Interval for $\mu$:
A bottle-filling machine is set to dispense 12.1 fluid ounces into juice bottles. To ensure that the machine is
filling accurately, every hour a worker randomly selects 15 bottles filled by the machine during the past hour
and measures the contents. If there is convincing evidence that the mean amount of juice dispensed is different
from 12.1 ounces or if there is convincing evidence that the standard deviation is greater than 0.05 ounce, the
machine is shut down for recalibration. It can be assumed that the amount of juice that is dispensed into bottles
is normally distributed.

During one hour, the number of fluid ounces of 15 randomly selected bottles was recorded below.

Construct a 95% confidence interval for the mean amount of juice dispensed by this machine during this hour.

In [2]:
bottle_sample = [12.33, 11.58, 12.14, 12.13, 11.96, 12.42, 12.0, 11.96, 11.73, 11.94, 11.74, 11.94, 11.66, 11.64, 11.94]

Calculate the T test statistic and the p-value needed for the t-test of significance. You can accomplish this however you like, but it is recommended you use the following method:

`statsmodels.stats.weightstats.DescrStatsW` (to use this method, run `from statsmodels.stats.weightstats import DescrStatsW` and call the method `DescrStatsW()`)

Save the bounds of the interval as a tuple to the variable `conf_int`.

In [3]:
from statsmodels.stats.weightstats import DescrStatsW
result = DescrStatsW(bottle_sample)
conf_int = result.tconfint_mean(.05)
print(f"The confidence interval is {conf_int}")

The confidence interval is (11.804607816653037, 12.076725516680296)


What conclusion would you make about the interval?
Be sure to use the context of the question in stating your conclusions!

__Answer:__

We are 95% confident that the true mean bottle fill for the machine during this hour is between 11.80 and 12.08oz.